src/Data/Vec/Properties.agda

```diff
@@ -1130,6 +1182,31 @@ toList∘fromList : (xs : List A) → toList (fromList xs) ≡ xs
 toList∘fromList List.[]       = refl
 toList∘fromList (x List.∷ xs) = cong (x List.∷_) (toList∘fromList xs)
 
+toList-cast : ∀ .(eq : m ≡ n) (xs : Vec A m) → toList (cast eq xs) ≡ toList xs
+toList-cast {n = zero}  eq []       = refl
+toList-cast {n = suc _} eq (x ∷ xs) =
+  cong (x List.∷_) (toList-cast (cong pred eq) xs)
+
+cast-fromList : ∀ {xs ys : List A} (eq : xs ≡ ys) →
+                cast (cong List.length eq) (fromList xs) ≡ fromList ys
+cast-fromList {xs = List.[]}     {ys = List.[]}     eq = refl
+cast-fromList {xs = x List.∷ xs} {ys = y List.∷ ys} eq = begin
+  x ∷ cast (cong (pred ∘ List.length) eq) (fromList xs) ≡⟨ cong (_ ∷_) (cast-fromList xs-equals-ys) ⟩
+  x ∷ fromList ys                                       ≡⟨ cong (_∷ _) x-equals-y ⟩
+  y ∷ fromList ys                                       ∎
+  where
+  x-equals-y = proj₁ (Listₚ.∷-injective eq)
```

Comment:
The names `x≡y` and `xs≡ys` would be shorter and more traditional. Also could we get away with `(x≡y, xs≡ys) = Listₚ.∷-injective eq`?

Reply:
Thanks for the suggestion! I changed them to irrefutable `let`.